packages/angular-cli/tasks/server/universal-dev-server.ts

```diff
@@ -0,0 +1,64 @@
+import * as path from 'path';
+const nodemon = require('nodemon');
+const tinyLr = require('tiny-lr');
+
+export class UniversalDevServer {
+  private appPath: string = '';
+  private nodemonConfig: any;
+  private demon: any;
+  private liveServer: any;
+
+  constructor(private webpackCompiler: any, private webpackDevServerConfiguration: any) {
+    this.appPath = path.join(
+      webpackDevServerConfiguration.contentBase,
+      `../dist/${webpackDevServerConfiguration.filename.replace('.ts', '.bundle.js')}`
+    );
+    this.liveServer = new tinyLr.Server();
+    this.liveServer.listen();
+  }
+
+  public listen(port: number, host: string, callBack: void) {
+    this.nodemonConfig = {
+      script: this.appPath,
+      runOnChangeOnly: true,
+      env: { 'PORT': port },
+      verbose: true,
+      ext: 'noop',
+      watch: ['noop/'],
+      ignore: ['*'],
+      stdout: false,
+      stderr: false
+    };
+
+    this.demon = nodemon(this.nodemonConfig);
+    this.demon
+      .on('restart', () => {
+        setTimeout(() => {
+          this.liveServer.changed({
+            body: {
+              files: ['LiveReload files']
+            }
+          });
+        }, 1000);
+      })
+      .on('stdout', (stdout: any) => {
+        console.log(stdout.toString().trim());
+      })
+      .on('stderr', (stderr: any) => {
+        console.log(stderr.toString());
+      });
+
+    this.webpackCompiler.plugin('compile', () => {});
```

Comment:
this technically should hang and break compiling because compile requires a callback which is passed in, in addition to the `compiler` instance. Why is this and the rest of these events being done.